### DISCUSSION TITLE
How to create multiple translation files inside a component ?

### DISCUSSION THREAD

**Discussion by yburakgazi** [WeblateOrg/weblate#4958] [Q&A]
Hello,

I'm interested in weblate and trying to create a demo for my managers, so we may proceed with weblate for multi-language support for the following projects. However, till now I couldn't achieve what I want to. I'm using resx files for multi-language support inside of a .Net Core application. Here are a couple of questions;

1) **Is a component responsible for translating only one resource file in a project ?**
- While creating a component, I'm forced to enter "Monolingual base language file" path. I can't skip it because the system throws an error message, "You can not use a monolingual translation without a base file.". I thought that "File Mask" pattern can find all resx file by using wildcard such as "Project/Resources/*.resx". There are many resx files under the folder Project/Resources. Tree structure of the resx files are such as Project/Resources/Controllers/Home/Index.en.resx, Project/Resources/Views/Home/Index.en.resx etc... Do I need to create several components for each resx file ? The problem is that in translation only Monolingual base language file is shown.

2) **How to create a new language using the same naming convention of the source file ?**
- When translation completes, I want to push the repository the created translation file using the same naming convention. For instance if the resource file is Project/Resources/Views/Home/Index.en.resx, the output should be Project/Resources/Views/Home/Index.cs.resx. I tried to rename from "Change translation" section on admin site ,however, at this time the system says "Filename Project/Resources/Views/Home/Index.cs.resx not found in repository! To add new translation, add language file into repository." I'm surprised to see this message because I thought that via weblate we can create a new resource file and push it into the repository. So, is it required to include all the translation files in the project ? I thought just having a resource file will be sufficient to convert it to the other languages.

**Comment by nijel** [comment]
> Is a component responsible for translating only one resource file in a project ?

Yes. In case you want to translate more files, you need more components. Use [component discovery](https://docs.weblate.org/en/latest/admin/addons.html#component-discovery) to create them automatically.

> How to create a new language using the same naming convention of the source file ?

Tools / Start new translation on component will add translation to new language named according to file mask and configured language code style, see https://docs.weblate.org/en/latest/admin/languages.html for more details.

**Comment by yburakgazi** [comment]
I have another question. I would like to continue from this discussion.

I started from scratch. I created a new project and a new component to make use of component discovery. After creating the first component, Addons was visible under Manage menu. I clicked addons and installed component discovery 4 times. In each component discovery instance, I was able to create multiple components using regular expression with the correct naming convention. Before clicking save button, I made sure that new components location match with tree structure of the resources files in the project. However, when I go to components section of the relavant project, I don't see any of them. I only see the base component's Monolingual base language file not the other ones. When I click Start Translation, the existing resx files are not received. I checked Component Discovery again regular expressions are okay. What am I doing wrong ?

**Reply by nijel** [reply]
The component discovery looks for existing files in the repository. If you started from scratch, there are none additional files.

**Reply by yburakgazi** [reply]
@nijel The files definitely exist in the repository. And Component Discovery can find them. But aftering saving component discovery configuration, I don't see them on weblate. How can i access them for translation ?

**Reply by nijel** [reply]
The creation is supposed to happen in the background, check whether celery is running and check its logs.

**Reply by yburakgazi** [reply]
Could you please tell me how can I check logs ?

**Reply by nijel** [reply]
I have no clue how you run Celery and where it is configured to store logs, sorry...

For generic hints, please see https://docs.weblate.org/en/latest/contributing/debugging.html#weblate-logs
